cmd/minikube/cmd/docker-env.go

```diff
@@ -381,14 +414,18 @@ docker-cli install instructions: https://minikube.sigs.k8s.io/docs/tutorials/doc
 			if err != nil {
 				exit.Error(reason.IfSSHClient, "Error with ssh-add", err)
 			}
-
 			cmd := exec.Command(path, d.GetSSHKeyPath())
 			cmd.Stderr = os.Stderr
+			cmd.Env = append(cmd.Env, fmt.Sprintf("SSH_AUTH_SOCK=%s", co.Config.SSHAuthSock))
+			cmd.Env = append(cmd.Env, fmt.Sprintf("SSH_AGENT_PID=%d", co.Config.SSHAgentPID))
 			err = cmd.Run()
 			if err != nil {
 				exit.Error(reason.IfSSHClient, "Error with ssh-add", err)
 			}
 		}
+
+		// eventually, run something similar to ssh --append-known
+		appendKnownHelper(nodeName, true)
```

Comment:
Not sure of a solution at the moment, the problem I see is lets say the user wants three terminal tabs they can use docker-env in, they'll run it three times, and it adds the same entry to `known_hosts` three times. Especially over time, this file could be filled with minikube docker-env entries.

Reply:
Indeed, but perhaps this is not a problem we can solve in this PR.

For example, when we shutdown the minikube, those items appended by minikube will still remain in known_hosts file, no matter whether we add `appendKnownHelper(nodeName, true)` or let the user manually execute ` minikube ssh-host --append-known`. No matter what kind of choices we make here, the problem that minikube leaves junk items in known_hosts will still exists.

Perhaps we may have a solution in the future, but I don't think it should be solved here.

Reply:
This change is the cause of https://github.com/kubernetes/minikube/issues/16933